### ISSUE TITLE
500 loading epub since 1.0.1

### ISSUE THREAD

**Issue by carlocapocasa** [Yetangitu/owncloud-apps#37]
Since updating to 1.0.1, when opening an epub, the server says in the log:

https://gist.github.com/carlocapocasa/0ae32ad963a7cc02a5e9d8fca44b084c

If you believe this is indeed a reader issue and you require further information please do let me know.

**Comment by Yetangitu** [comment]
Do you have OPDS Catalog installed?

**Comment by carlocapocasa** [comment]
No, I don't. I installed it and the problem went away.

**Comment by Yetangitu** [comment]
It should not have given that message though... so while your issue went away I still have to check why this happens. Reader does check for the presence of the class `\OCA\Files_Opds\Meta` before trying to use any related functionality:

```php
    public function get($fileId, $name=null) {
        if (class_exists('\OCA\Files_Opds\Meta')) {
            ...do things with Meta::get()...
```

...but that check should not trigger these messages...

**Comment by carlocapocasa** [comment]
OK thanks for the workaround.

From briefly examining the error, it appears that $validRoots in lib/autoloader.php does not contain the appropriate roots to load meta.php at the time Autoloader::load is getting called. I would guess that loading happens before calling ::get on my system, perhaps due to differences in opcode caching in PHP 7.0 as opposed to 5.x APC?

**Comment by Yetangitu** [comment]
Well, I'm running PHP 7.0 and I don't see any errors so I guess it is not a 5-vs-7 problem.

**Comment by Yetangitu** [comment]
Looking through the NC source I find more occurrences of autoloader-related problems (e.g. https://github.com/nextcloud/server/blob/master/lib/private/Server.php#L812). I'll check if `$c->getAppManager()->isInstalled('files_opds')` does the job and use that if it does.

**Comment by Yetangitu** [comment]
Fixed in [v1.0.2](a8a141946c63d2eb756c8ac51826d3ee0e524499)
